From d473f1e1f6ba80bfaee4daa058da159305167323 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Thu, 12 Dec 2024 04:16:02 -0500
Subject: [PATCH] Update declarations to allow compile with gcc-15

This patch fixes some of the compile errors with gcc 15-20241117.

In addition the follow declarations need to be fixed:
  sed -n 75,77p libtirpc-1.3.6/src/key_call.c
  cryptkeyres *(*__key_encryptsession_pk_LOCAL)() = 0;
  cryptkeyres *(*__key_decryptsession_pk_LOCAL)() = 0;
  des_block *(*__key_gendes_LOCAL)() = 0;

Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>
Signed-off-by: Steve Dickson <steved@redhat.com>
---
 src/auth_none.c     | 2 +-
 src/getpublickey.c  | 2 +-
 src/svc_auth_none.c | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

--- a/src/auth_none.c
+++ b/src/auth_none.c
@@ -62,7 +62,7 @@ static bool_t authnone_validate (AUTH *,
 static bool_t authnone_refresh (AUTH *, void *);
 static void authnone_destroy (AUTH *);
 
-extern bool_t xdr_opaque_auth();
+extern bool_t xdr_opaque_auth(XDR *, struct opaque_auth *);
 
 static struct auth_ops *authnone_ops();
 
--- a/src/getpublickey.c
+++ b/src/getpublickey.c
@@ -52,7 +52,7 @@
 /*
  * Hack to let ypserv/rpc.nisd use AUTH_DES.
  */
-int (*__getpublickey_LOCAL)() = 0;
+int (*__getpublickey_LOCAL)(const char *, char *) = 0;
 
 /*
  * Get somebody's public key
--- a/src/svc_auth_none.c
+++ b/src/svc_auth_none.c
@@ -37,8 +37,8 @@
 
 #include <rpc/rpc.h>
 
-static bool_t	svcauth_none_destroy();
-static bool_t   svcauth_none_wrap();
+static bool_t	svcauth_none_destroy(SVCAUTH *);
+static bool_t   svcauth_none_wrap(SVCAUTH *, XDR *, bool_t (*)(XDR *, ...), char *);
 
 struct svc_auth_ops svc_auth_none_ops = {
 	svcauth_none_wrap,
